extensions/wikia/Email/Controller/FacebookDisconnectController.class.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\EmailController;
+use Email\Tracking\TrackingCategories;
+
+class FacebookDisconnectController extends ForgotPasswordController {
+	const TRACKING_CATEGORY = TrackingCategories::TEMPORARY_PASSWORD;
+
+	public function getSubject() {
+		return $this->getMessage( 'emailext-fbdisconnect-subject' )->text();
+	}
+
+	/**
+	 * @template temporaryPassword
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'passwordIntro' => $this->getIntro(),
+			'username' => $this->getMessage( 'emailext-fbdisconnect-username',
+				$this->targetUser->getName() )->parse(),
```

Comment:
You only need ```parse``` if the message contains wikitext.  For simple substitutions (e.g. ```$1```, ```$2```), ```text()``` will suffice.  This has the advantage of not firing up the parser for the string.